test/spec/RunSpec.hs

```diff
@@ -60,6 +60,40 @@ spec =
           stdout output `shouldBe` "Hello, world!\n"
           exitCode output `shouldBe` ExitSuccess
 
+        it "runs non-default executables within projects" $ do
+          writeFile
+            "garn.ts"
+            [i|
+              import * as garn from "#{repoDir}/ts/mod.ts"
+              export const project = garn.mkProject({
+                description: "my project",
+                defaultEnvironment: garn.emptyEnvironment,
+              }, {}).addExecutable("hello")`echo Hello, world!`;
+            |]
+          output <- runGarn ["run", "project/hello"] "" repoDir Nothing
+          stdout output `shouldBe` "Hello, world!\n"
+          exitCode output `shouldBe` ExitSuccess
+
+        it "allows specifying deeply nested executables" $ do
+          writeFile
+            "garn.ts"
+            [i|
+              import * as garn from "#{repoDir}/ts/mod.ts"
+              const a = garn.mkProject({
+                description: "a",
+                defaultExecutable: garn.shell`echo executable in a`,
+              }, {});
+              const b = garn.mkProject({
+                description: "b",
+              }, { a });
+              export const c = garn.mkProject({
+                description: "b",
+              }, { b });
+            |]
+          output <- runGarn ["run", "c/b/a"] "" repoDir Nothing
```

Comment:
Can't we just replace dots with slashes, so the *interface* still uses dots? (It doesn't matter *that* much that the flake file is different)